lib/src/merged_tree.rs

```diff
@@ -274,14 +284,6 @@ pub struct TreeDiffEntry {
 /// ones) can fetch trees asynchronously.
 pub type TreeDiffStream<'matcher> = BoxStream<'matcher, TreeDiffEntry>;
 
-fn all_tree_basenames(trees: &Merge<Tree>) -> impl Iterator<Item = &RepoPathComponent> {
-    trees
-        .iter()
-        .map(|tree| tree.data().names())
-        .kmerge()
-        .dedup()
-}
-
```

Comment:
nit: deleting this seems to be an unrelated cleanup that should ideally be in its own commit